smart_fact_crawler/__init__.py

```diff
@@ -39,252 +43,306 @@ def smartfact(timeout=None):
+        'control_deviation': Quantity(s2f(get(table, 6, 1)), 'arcsec'),
+        'moon_distance': Quantity(s2f(get(table, 7, 1)), 'deg'),
     })
 
 
-def drive_pointing(url=None, timeout=None):
+def drive_pointing(url=None, timeout=None, fallback=False):
     if url is None:
         url = os.path.join(smartfacturl, 'pointing.data')
+
     table = smartfact2table(url, timeout=timeout)
+    get = partial(get_entry, fallback=fallback)
+
     return to_namedtuple('PointingPage', {
-        'timestamp': sft2dt(table[0][0]),
-        'azimuth': Quantity(s2f(table[1][1]), 'deg'),
-        'zenith_distance': Quantity(s2f(table[2][1]), 'deg'),
+        'timestamp': sft2dt(get(table, 0, 0)),
+        'azimuth': Quantity(s2f(get(table, 1, 1)), 'deg'),
+        'zenith_distance': Quantity(s2f(get(table, 2, 1)), 'deg'),
     })
 
 
-def sqm(url=None, timeout=None):
+def sqm(url=None, timeout=None, fallback=False):
     if url is None:
         url = os.path.join(smartfacturl, 'sqm.data')
+
     table = smartfact2table(url, timeout=timeout)
+    get = partial(get_entry, fallback=fallback)
+
     return to_namedtuple('SqmPage', {
-        'timestamp': sft2dt(table[0][0]),
-        'magnitude': Quantity(s2f(table[1][1]), 'mag'),
-        'sensor_frequency': Quantity(s2f(table[2][1]), 'Hz'),
-        'sensor_period': Quantity(s2f(table[4][1]), 's'),
-        'sensor_temperature': Quantity(s2f(table[5][1]), 'deg_C'),
+        'timestamp': sft2dt(get(table, 0, 0)),
+        'magnitude': Quantity(s2f(get(table, 1, 1)), 'mag'),
+        'sensor_frequency': Quantity(s2f(get(table, 2, 1)), 'Hz'),
+        'sensor_period': Quantity(s2f(get(table, 4, 1)), 's'),
+        'sensor_temperature': Quantity(s2f(get(table, 5, 1)), 'deg_C'),
     })
 
 
-def sun(url=None, timeout=None):
+def sun(url=None, timeout=None, fallback=False):
     if url is None:
         url = os.path.join(smartfacturl, 'sun.data')
 
     def next_datetime_from_hhmm_string(hhmm):
+        if hhmm is None:
+            return None
+
         now = datetime.utcnow()
         hour = int(hhmm[0:2])
         minute = int(hhmm[3:5])
 
         new_date = now.replace(hour=hour, minute=minute, second=0)
         if not new_date > now:
             new_date += timedelta(days=1)
+
         return new_date
 
     conv = next_datetime_from_hhmm_string
 
     table = smartfact2table(url, timeout=timeout)
+    get = partial(get_entry, fallback=fallback)
+
     return to_namedtuple('SunPage', {
-        'timestamp': sft2dt(table[0][0]),
-        'end_of_dark_time': conv(table[1][1]),
-        'end_of_astronomical_twilight': conv(table[2][1]),
-        'end_of_nautical_twilight': conv(table[3][1]),
-        'start_of_day_time': conv(table[4][1]),
-        'end_of_day_time': conv(table[5][1]),
-        'start_of_nautical_twilight': conv(table[6][1]),
-        'start_of_astronomical_twilight': conv(table[7][1]),
-        'start_of_dark_time': conv(table[8][1]),
+        'timestamp': sft2dt(get(table, 0, 0)),
```

Comment:
That would adding a new class, which I didn't think about. One could do that, but I'm also fine with a function that acts on built in lists